### PR TITLE
Pelorus webhook exporter doc

### PR DESCRIPTION
**Pull request by mpryc**
Updated documentation to cover usage of the pelorus webhook exporter with the examples on how to use
curl in a sample application lifecycle.

Fixes: #857 

@redhat-cop/mdt

### CODE REVIEW

File: docs/Development.md [hunk 1]
```diff
@@ -404,8 +411,14 @@ Running an exporter on your local machine should follow this process:
 
 At this point, your exporter should be available at http://localhost:8080
 
-    curl http://localhost:8080
+    curl http://localhost:8080/metrics
+
+7. (Optional) Webhook exporter
+
+Webhook type exporter has an additional URL target http://localhost:8080/pelorus/webhook, that allows to receive POST data, for example using curl:
```
Comment (mateusoliveira43) [inline docs/Development.md:418]: to run this command, user does not have to be at `exporters/tests/data`? I think it is good to add this info

File: docs/GettingStarted/Examples.md [hunk 2]
```diff
@@ -196,3 +196,21 @@ spec:
           - name: JIRA_RESOLVED_STATUS
             value: DONE
 ```
+
+## Webhook exporter only
+
+In this example, Pelorus will receive data from the external systems without making any API calls to OCP nor external providers.
```
Comment (mateusoliveira43) [inline docs/GettingStarted/Examples.md:202]: OpenShift instead of OCP? (I get confused with these abbreviations)

File: docs/Architecture.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ Pelorus is composed of the following open source components.
     * Commit Time
     * Deploy Time
     * Failure
+    * Webhook Exporter
```
Comment (mateusoliveira43) [inline docs/Architecture.md:18]: just Webhook? this bullet is already for exporters

File: docs/GettingStarted/configuration/ExporterWebhook.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
+# Webhook Exporter
+
+The Webhook exporter is capable of receiving data from any service that sends an appropriate HTTP POST request.
+
+The HTTP POST request consists of two parts, [Header and the Payload](#webhook-headers-and-payloads). The Header data specifies which webhook plugin should be used to validate and transform the payload into a Pelorus metric, that is consumed by Prometheus.
```
Comment (mateusoliveira43) [inline docs/GettingStarted/configuration/ExporterWebhook.md:5]: isn't "Model" instead of "plugin" in this line?

File: docs/GettingStarted/configuration/ExporterWebhook.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+The payload data received by the webhook exporter may be any type of data that other exporters has to pull from other providers or OCP objects, such as [Failure](./ExporterFailure.md), [Commit Time](./ExporterCommittime.md), [Deploy Time](./ExporterDeploytime.md) and more.
```
Comment (mateusoliveira43) [inline docs/GettingStarted/configuration/ExporterWebhook.md:7]: I would rephrase here. something like "The payload data received by the webhook exporter may be of the following types: <link to the types under payload-specification section>"

File: docs/GettingStarted/configuration/ExporterWebhook.md (hunk 4 above)
Comment (mpryc) [reply docs/GettingStarted/configuration/ExporterWebhook.md:5]: it should be "plugin". The "Model" is an internal implementation detail here that applies to the plugin.

File: docs/GettingStarted/configuration/ExporterWebhook.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+## Webhook headers and payloads
+
+When sending an HTTP POST request to the webhook's configured URL endpoint, the payload must conform to the webhook payload specification and include several special headers. It's important to note that the header specifications may vary depending on the Pelorus plugin determined by the `User-Agent` Header value and are described per plugin, alongside the payload specification.
+
+### `Pelorus-Webhook/`
+#### Headers specification
+
+| Header | Description |
+|--------|---------------|
+| `User-Agent` | Prefix must begin with `Pelorus-Webhook/` in order to use this plugin. |
```
Comment (mateusoliveira43) [inline docs/GettingStarted/configuration/ExporterWebhook.md:71]: did not understand this. I can put anything after and it works?

File: docs/Development.md (hunk 1 above)
Comment (mpryc) [reply docs/Development.md:418]: fixed.

File: docs/GettingStarted/Examples.md (hunk 2 above)
Comment (mpryc) [reply docs/GettingStarted/Examples.md:202]: changed.

File: docs/Architecture.md (hunk 3 above)
Comment (mpryc) [reply docs/Architecture.md:18]: changed.

File: docs/GettingStarted/configuration/ExporterWebhook.md (hunk 5 above)
Comment (mpryc) [reply docs/GettingStarted/configuration/ExporterWebhook.md:7]: We may and will have other type of metrics (SLA/SLO as an example).

**Comment by mateusoliveira43** [comment]
update webhook REDAME (8000 to 8080)
```diff
-$ curl -X POST http://localhost:8000/pelorus/webhook -d @./webhook_pelorus_committime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: committime"
-$ curl -X POST http://localhost:8000/pelorus/webhook -d @./webhook_pelorus_deploytime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: deploytime"
-$ curl -X POST http://localhost:8000/pelorus/webhook -d @./webhook_pelorus_failure_created.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: failure"
-$ curl -X POST http://localhost:8000/pelorus/webhook -d @./webhook_pelorus_failure_resolved.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: failure"
+$ curl -X POST http://localhost:8080/pelorus/webhook -d @./webhook_pelorus_committime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: committime"
+$ curl -X POST http://localhost:8080/pelorus/webhook -d @./webhook_pelorus_deploytime.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: deploytime"
+$ curl -X POST http://localhost:8080/pelorus/webhook -d @./webhook_pelorus_failure_created.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: failure"
+$ curl -X POST http://localhost:8080/pelorus/webhook -d @./webhook_pelorus_failure_resolved.json -H "Content-Type: application/json" -H "User-Agent: Pelorus-Webhook/test" -H "X-Pelorus-Event: failure"
 
-Navigate to the endpoint [http://localhost:8000/metrics](http://localhost:8000/metrics), you should see all the metrics collected:
+Navigate to the endpoint [http://localhost:8080/metrics](http://localhost:8080/metrics), you should see all the metrics PE commit_timestamp gauge
```

File: docs/GettingStarted/configuration/ExporterWebhook.md (hunk 6 above)
Comment (mpryc) [reply docs/GettingStarted/configuration/ExporterWebhook.md:71]: Yes, the spec normally have just prefix, so it's same to have user-agent:

`Pelorus-Webhook/something`
`Pelorus-Webhook/something2`

Similarly it's for other services such as GitHub, please check last sentence in the paragraph:
https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#delivery-headers

Comment (mateusoliveira43) [reply docs/GettingStarted/configuration/ExporterWebhook.md:71]: hmm it works like and id then. ok

**Review by mateusoliveira43 — APPROVED** [review]
